pkg/runctl/controller.go

```diff
@@ -242,17 +249,10 @@ func (c *Controller) syncHandler(key string) error {
 		return nil
 	}
 
-	// the configuration should be loaded once per sync to avoid inconsistencies
-	// in case of concurrent configuration changes
-	pipelineRunsConfig, err := loadPipelineRunsConfig(c.factory)
-	if err != nil {
-		return errors.Wrap(err, "failed to load configuration for pipeline runs")
-	}
-
 	// Check if object has deletion timestamp
 	// If not, try to add finalizer if missing
 	if pipelineRun.HasDeletionTimestamp() {
-		runManager := c.createRunManager(pipelineRun, pipelineRunsConfig)
+		runManager := c.createRunManager(pipelineRun, nil)
```

Comment:
Passing `nil` as configuration to `newRunManager()` with the assumption that subsequent calls of methods on the returned instance do not need it is questionable. Actually `newRunManager()` would need to fail if required parameter are nil (or otherwise invalid).

Change `runManager.Start()` to take the configuration as parameter, and do not store it in the `runManager` struct anymore. Then it's clear that `runManager.Start()` requires a configuration, but `runManager.Cleanup()` does not.

Reply:
Done